### PR TITLE
Add MAV_MOUNT_MODE_SYSID_TARGET and MAV_CMD_DO_SET_ROI_SYSID

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

**Comment by hamishwillee** [comment]
@julianoes This looks fine to me. Any objection?

**Comment by hamishwillee** [comment]
Hmmm. The command/enums themselves seem pretty obvious - this sets the camera/vehicle to track an ROI that is a mavlink system with a specific ID. 

@peterbarker How is the camera supposed to get the position of that ID? @julianoes What other questions did you have (specifically)?

**Comment by peterbarker** [comment]
On Wed, 27 Nov 2019, Hamish Willee wrote:

> @peterbarker How is the camera supposed to get the position of that ID? @julianoes What other questions did you have (specifically)?

That's deliberately not specified :-)

At the moment the obvious method is GLOBAL_POSITION_INT - and that's what 
I'm currently using.  An autopilot may also wish to extrapolate based on 
velocity (also from GLOBAL_POSITION_INT) and acceleration (from somewhere 
else....).

However, that's assuming a global position, which may or may not be valid, 
depending on circumstance.  A LOCAL_POSITION_NED might also be a useful 
thing to usem for example.

I should have specified that the method use to get location is 
deliberately unspecified ("This page intentionally left blank", "Left as 
an exercise to the coder", ...)

File: message_definitions/v1.0/common.xml [hunk 1]
```diff
@@ -990,6 +990,9 @@
       <entry value="4" name="MAV_MOUNT_MODE_GPS_POINT">
         <description>Load neutral position and start to point to Lat,Lon,Alt</description>
       </entry>
+      <entry value="5" name="MAV_MOUNT_MODE_SYSID_TARGET">
+        <description>Follow system ID</description>
```
Comment (hamishwillee) [inline message_definitions/v1.0/common.xml]: Perhaps:

> Gimbal tracks system with specified system ID

File: message_definitions/v1.0/common.xml [hunk 2]
```diff
@@ -1562,6 +1565,10 @@
         <param index="6">Empty</param>
         <param index="7">Empty</param>
       </entry>
+      <entry value="198" name="MAV_CMD_DO_SET_ROI_SYSID">
+        <description>Camera ROI is vehicle with specified SysID.</description>
```
Comment (hamishwillee) [inline message_definitions/v1.0/common.xml]: Perhaps:

> Camera tracks system with specified system ID. Position of target system is obtained out-of-channel.

**Comment by hamishwillee** [comment]
@peterbarker I've proposed some words that make that explicit - reasonable?

**Comment by peterbarker** [comment]
On Thu, 28 Nov 2019, Hamish Willee wrote:

>       Camera tracks system with specified system ID. Position of target system is obtained out-of-channel.

It's not out-of-channel as "channel" is most likely to be interpreted as 
"mavlink channel" :-)

Also, this is a mount, not necessarily wielding a camera.

Perhaps:

"Mount tracks system with specified system ID.  Determination of position 
of target vehicle may be done with GLOBAL_POSITION_INT or any other 
means."

Provides a hint but doesn't limit.

**Comment by hamishwillee** [comment]
@peterbarker That works for me. Thanks.

**Pull request merged**